pkg/gitclient/helpers.go

```diff
@@ -354,6 +354,38 @@ func CloneToDir(g Interface, gitURL, dir string) (string, error) {
 	return dir, nil
 }
 
+// PartialCloneToDir Partially clones the git repository to either the given directory or create a temporary one
+// Alternative to SparseCloneToDir when git provider does not support sparse-checkout
+// sparseCheckoutPatterns not supported
+// If shallow is true the clone is made with --depth=1
+func PartialCloneToDir(g Interface, gitURL, dir string, shallow bool) (string, error) {
+	var err error
+	if dir != "" {
+		err = os.MkdirAll(dir, util.DefaultWritePermissions)
+		if err != nil {
+			return "", fmt.Errorf("failed to create directory %s: %w", dir, err)
+		}
+	} else {
+		dir, err = os.MkdirTemp("", "jx-git-")
+		if err != nil {
+			return "", fmt.Errorf("failed to create temporary directory: %w", err)
+		}
+	}
```

Comment:
We can pull this out into another private function as its being reused by `CloneToDir()`

Reply:
Done :)